flocker/provision/_ssh/_crochet.py

```diff
@@ -0,0 +1,28 @@
+from crochet import setup, run_in_reactor
+from effect import ComposedDispatcher
+
+from .._effect import dispatcher
+from ._conch import dispatcher as conch_dispatcher
+from effect.twisted import (
+    make_twisted_dispatcher,
+    perform as perform_with_twisted
+)
+from twisted.internet import reactor
+
+
+def perform(dispatcher, effect):
+    """
+    Perform an effect in a reactor with crochet.
+    """
+    setup()
+    return run_in_reactor(perform_with_twisted)(
```

Comment:
It's not clear what's going on here. Maybe use keyword arguments and assign the returned function to a descriptive variable before calling it.